pipelines/zendesk/__init__.py

```
@@ -0,0 +1,221 @@
+    @:param start_date: Datetime object that serves as the starting date of when to retrieve objects
+    @:returns: Generator of dicts
+    """
+
+    try:
+        # use the zenpy endpoint with methods to yield the data
+        resource_api_method = getattr(zendesk_client, f"{resource}")
+        resource_api_method_incremental = resource_api_method.incremental
+        resource_api = resource_api_method_incremental(start_date=start_date, per_page=per_page)
+    except TypeError:
+        # this resource doesn't support incremental so just load normally
+        resource_api_method = getattr(zendesk_client, f"{resource}")
+        resource_api = resource_api_method()
+    except Exception as e:
+        logger.warning(f"Skipped resource: {resource}. Couldn't load!")
+        logger.warning(str(e))
+        return
+    dict_generator = basic_load(resource_api=resource_api)
+    yield from dict_generator
+
+
+def talk_resource(zendesk_client: ZendeskAPIClient, talk_endpoint_name: str, talk_endpoint: str) -> Iterator[TDataItem]:
+    """
+    Called as a dlt resource. Generic loader for a ZendeskTalk endpoint.
+    @:param zendesk_client: ZendeskAPIClient object that makes api calls to Zendesk Talk API
+    @:param talk_endpoint_name: The name of the talk_endpoint
+    @:param talk_endpoint: The actual url ending of this endpoint
+    @:returns: Generator of dicts
+    """
+    try:
+        my_req = zendesk_client.make_request(endpoint=talk_endpoint, data_point_name=talk_endpoint_name)
+        if my_req:
+            my_pages = [page for page in my_req]
+            for page in my_pages:
+                # multiple records
+                if isinstance(page, list):
+                    for record in page:
+                        yield record
+                else:
+                    yield page
+    except Exception as e:
+        logger.warning(f"Failed to load resource: {talk_endpoint}")
```

swallowing exceptions looks really dangerous. you should log and then raise